### ISSUE TITLE
Bug: sam local invoke -d  option now working with nodejs16.x

### ISSUE THREAD

**Issue by deepak-panwar-paf** [aws/aws-sam-cli#3886]
### Description:
sam local invoke -d 5858 <Function name> -e <Event path> is failing for nodejs16.x it working fine in nodejs14.x

### Steps to reproduce:
create `nodejs16.x`  `Hello World Example` project  project from `sam init` command
run command `sam local invoke -d 5858 HelloWorldFunction -e events/event.json`

inside vs code inspect using following launch configuration

```json
{
            "name": "inspector",
            "type": "node",
            "request": "attach",
            "address": "localhost",
            "port": 5858,
            "localRoot": "${workspaceRoot}",
            "remoteRoot": "/var/task",
            "protocol": "inspector",
            "stopOnEntry": false
        }
```

### Observed result:
START RequestId: ff3620cf-03d2-4a3f-a204-9d4665c56dd5 Version: $LATEST
Debugger listening on ws://0.0.0.0:5858/ecdab66e-0a85-4da9-b4ec-8e0e511faeaa
For help, see: https://nodejs.org/en/docs/inspector
Debugger attached.
Waiting for the debugger to disconnect...
node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module '/var/runtime/index.js'
    at Function._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function._load (node:internal/modules/cjs/loader:778:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
    at node:internal/main/run_main_module:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

### Observations
I checked in the docker image there is no such file with name `index.js` inside `var/task` folder in node16 image only `index.mjs` file was present. `index.js` file is present in node14 docker image

### Expected result:
With given launch configuration you should be able to inspect in the source code instead of getting error

Additional environment details (Ex: Windows, Mac, Amazon Linux etc)
OS: Mac
sam --version: 1.50.0

**Comment by torresxb1** [comment]
Thanks for bringing this to our attention, we will be investigating it further. Please watch this channel for more updates, and feel free to reach out.

**Comment by mndeveci** [comment]
This is been released with [SAM CLI v1.51.0](https://github.com/aws/aws-sam-cli/releases/tag/v1.51.0)

**Comment by Leslie-Wong-H** [comment]
Directly migrating from 12.x to 16.x using the sam template seemingly would result in the loss of env at lambda runtime. Suffered from this for a while, I finally figured it out and add the env back. Take care.
